llvm/include/llvm/CodeGen/GlobalISel/Utils.h

```diff
@@ -593,5 +593,27 @@ bool isGuaranteedNotToBeUndef(Register Reg, const MachineRegisterInfo &MRI,
 /// estimate of the type.
 Type *getTypeForLLT(LLT Ty, LLVMContext &C);
 
+enum class GIConstantKind { Scalar, FixedVector, ScalableVector };
+
+/// An integer-like constant.
```

Comment:
This class could do with more documentation about its intended purpose, in what situation it's supposed to be used.